viewing_party/party.py

```diff
@@ -1,23 +1,137 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if (not title) or (not genre) or (not rating):
+        return None
+      
+    if not all((isinstance(title, str), isinstance(genre, str),
+               isinstance(rating, float))):
+        return None
+
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    # Handles if create_movie() returns None for movie
+    if movie:
+        user_data["watched"].append(movie)
+       
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    if movie:
+        user_data["watchlist"].append(movie)
+   
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            move_movie = movie
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(move_movie)
+
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    sum = 0
+    if not user_data["watched"]:
+        return 0.0
+
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+
+    return sum/len(user_data["watched"])
+
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+
+    genre_list = [movie["genre"] for movie in user_data["watched"]]
+   
+    # Returns the first instance of a most-watched genre
+    return max(genre_list, key=genre_list.count)
+
+
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+
+def get_unique_watched(user_data):
+    friend_watched = {movie_dict["title"] for friend_dict in user_data["friends"]
+                      for movie_dict in friend_dict["watched"]}
+
+    user_watched = [user_dict for user_dict in user_data["watched"]
+                    if user_dict["title"] not in friend_watched]

```

Comment:
I think we could go one step further and separate these data structures into their own helper functions. They can then be called below in `get_friends_unique_watched` to cut down on repetitive loops.

We'd have one that flattens the friends' movies watched, and one that cleans up the user's watched list or unique watched. Or create a helper function for both!